lib/parlour/type_parser.rb

```diff
@@ -0,0 +1,609 @@
+# typed: true
+
+# TODO: support sig without runtime
+
+# Suppress versioning warnings - the majority of users will not actually be
+# using this, so we don't want to pollute their console
+old_verbose = $VERBOSE
+begin
+  $VERBOSE = nil
+  require 'parser/current'
+ensure
+  $VERBOSE = old_verbose
+end
+
+module Parlour
+  # Parses Ruby source to find Sorbet type signatures.
+  class TypeParser
+    # Represents a path of indeces which can be traversed to reach a specific
+    # node in an AST.
+    class NodePath
+      extend T::Sig
+
+      sig { returns(T::Array[Integer]) }
+      # @return [Array<Integer>] The path of indeces.
+      attr_reader :indeces
```

Comment:
I think this should be `indices`? As in the plural of index?

Reply:
Oops, you're right 🙃 Thanks!